packages/compass-e2e-tests/tests/collection-aggregations-tab.test.ts

```diff
@@ -329,13 +331,19 @@ describe('Collection aggregations tab', function () {
     await waitForAnyText(browser, await browser.$(Selectors.stageContent(1)));
 
     // make sure it complains that it must be the last stage
-    const messageElement = await browser.$(
-      Selectors.stageEditorErrorMessage(1)
-    );
-    await messageElement.waitForDisplayed();
-
-    expect(await messageElement.getText()).to.equal(
-      '$out can only be the final stage in the pipeline'
+    await browser.waitUntil(
+      async () => {
+        const messageElement = await browser.$(
+          Selectors.stageEditorErrorMessage(1)
+        );
+        await messageElement.waitForDisplayed();
+        const text = await messageElement.getText();
+        return text === '$out can only be the final stage in the pipeline';
+      },
+      {
+        timeoutMsg:
+          'Waited for the error "$out can only be the final stage in the pipeline"',
```

Comment:
I dislike having to use waitUntil unless it is absolutely necessary because then you have to write a timeout message every time to get useful test output on failure. But oh well.